src/coreComponents/python/modules/geosx_mesh_doctor/checks/vtk_polyhedron.py

```diff
@@ -0,0 +1,202 @@
+
+
+class FaceStream:
+    """
+    Helper class to manipulate the vtk face streams.
+    """
+    def __init__(self, data: Tuple[Tuple[int, ...], ...]):
+        # self.__data contains the list of faces nodes, like it appears in vtk face streams.
+        # Except that the additional size information is removed
+        # in favor of the __len__ of the containers.
+        self.__data = data
+
+    @staticmethod
+    def build_from_vtk_id_list(ids: vtkIdList):
+        """
+        Builds a FaceStream from the raw vtk face stream.
+        :param ids: The vtk face stream.
+        :return: A new FaceStream instance.
+        """
+        return FaceStream(parse_face_stream(ids))
+
+    @property
+    def face_nodes(self) -> Iterable[Sequence[int]]:
+        """
+        Iterate on the nodes of all the faces.
+        :return: An iterator.
+        """
+        return iter(self.__data)
+
+    @property
+    def num_faces(self) -> int:
+        """
+        Number of faces in the face stream
+        :return: An integer
+        """
+        return len(self.__data)
+
+    @property
+    def support_point_ids(self) -> Iterable[int]:
+        """
+        The list of all (unique) support points of the face stream, in no specific order.
+        :return: The set of all the point ids.
+        """
+        tmp = []
+        for nodes in self.face_nodes:
+            tmp += nodes
+        return frozenset(tmp)
+
+    @property
+    def num_support_points(self) -> int:
+        """
+        The number of unique support nodes of the polyhedron.
+        :return: An integer.
+        """
+        return len(self.support_point_ids)
+
+    def __getitem__(self, face_index) -> Tuple[int, ...]:
+        """
+        The support point ids for the `face_index` face.
+        :param face_index: The face index (within the face stream).
+        :return: A tuple containing all the point ids.
+        """
+        return self.__data[face_index]
+
+    def flip_faces(self, face_indices):
+        """
+        Returns a new FaceStream instance with the face indices defined in face_indices flipped.,
+        :param face_indices: The faces (local) indices to flip.
+        :return: A newly created instance.
+        """
+        result = []
+        for face_index, face_nodes in enumerate(self.__data):
+            result.append(tuple(reversed(face_nodes)) if face_index in face_indices else face_nodes)
+        return FaceStream(tuple(result))
+
+    def dump(self) -> Sequence[int]:
+        """
+        Returns the face stream awaited by vtk, but in a python container.
+        The content can be used, once converted to a vtkIdList, to define another polyhedron in vtk.
+        :return: The face stream in a python container.
+        """
+        result = [len(self.__data)]
+        for face_nodes in self.__data:
+            result.append(len(face_nodes))
+            result += face_nodes
+        return tuple(result)
+
+    def __repr__(self):
```

Comment:
- `repr` are standard attributes and methods.
- `__repr` is a naming convention for `private` members in python. The mangling is modified (see [here](https://www.tutorialsteacher.com/python/public-private-protected-modifiers) for example).
- Attributes with double underscores before and after are meant to be "**special**". For [example](https://docs.python.org/3/library/operator.html), `__add__` will define the `+`, `__eq__` the `==`, `__init__` is the constructor and here `__repr__` is meant to return a printable `repr`esentation of the instance (this will be called when you call `print(instance)`). Thus, the name is important here and cannot be changed.